File: apps/dotnet-service/src/APIs/Room/Dtos/RoomWhereInput.cs
using DotnetService.Core.Enums;

namespace DotnetService.APIs.Dtos;

public class RoomWhereInput
{
    public List<string>? Bookings { get; set; }

    public DateTime? CreatedAt { get; set; }

    public string? Description { get; set; }

    public string? Hotel { get; set; }

    public string? Id { get; set; }

    public string? Name { get; set; }

    public double? Price { get; set; }

    public string? RoomNumber { get; set; }

    public TypeFieldEnum? TypeField { get; set; }

    public DateTime? UpdatedAt { get; set; }
}

File: apps/hotel-booking-service-admin/src/api/room/RoomWhereInput.ts
import { BookingListRelationFilter } from "../booking/BookingListRelationFilter";
import { StringNullableFilter } from "../../util/StringNullableFilter";
import { HotelWhereUniqueInput } from "../hotel/HotelWhereUniqueInput";
import { StringFilter } from "../../util/StringFilter";
import { FloatNullableFilter } from "../../util/FloatNullableFilter";

export type RoomWhereInput = {
  bookings?: BookingListRelationFilter;
  description?: StringNullableFilter;
  hotel?: HotelWhereUniqueInput;
  id?: StringFilter;
  name?: StringNullableFilter;
  price?: FloatNullableFilter;
  roomNumber?: StringNullableFilter;
  typeField?: "Option1";
};
